dot/core/service.go

```diff
@@ -346,15 +352,17 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 				continue
 			}
 
-			externalExt := make(types.Extrinsic, 0, 1+len(ext))
-			externalExt = append(externalExt, byte(types.TxnExternal))
-			externalExt = append(externalExt, ext...)
-			txv, err := rt.ValidateTransaction(externalExt)
+			externalExt, err := s.buildExternalTransaction(rt, ext)
+			if err != nil {
+				return fmt.Errorf("building external transaction: %w", err)
+			}
+
+			transactionValidity, err := rt.ValidateTransaction(externalExt)
 			if err != nil {
 				logger.Debugf("failed to validate transaction for extrinsic %s: %s", ext, err)
 				continue
```

Comment:
nit a bit out of scope, but shouldn't we return the error?

Reply:
seems to me no. Since we have to check every extrinsic, doesn't make sense to me that one invalid one should halt that process. Feels like attack vector. Maybe there is some other appropriate way to handle though.